src/display/dom_utils.js

```diff
@@ -33,6 +33,36 @@ var createValidAbsoluteUrl = sharedUtil.createValidAbsoluteUrl;
 
 var DEFAULT_LINK_REL = 'noopener noreferrer nofollow';
 
+function DOMCanvasFactory() {}
+DOMCanvasFactory.prototype = {
+  create: function DOMCanvasFactory_create(width, height) {
+    this.canvas = document.createElement('canvas');
+    this.canvas.width = width || 0;
+    this.canvas.height = height || 0;
+
+    this.destroyed = false;
+    return this.canvas;
+  },
+
+  reset: function DOMCanvasFactory_reset(width, height) {
+    if (this.destroyed) {
+      throw new Error('Canvas not found');
+    }
+    this.canvas.width = width;
+    this.canvas.height = height;
+
+    return this.canvas;
+  },
+
+  destroy: function DOMCanvasFactory_destroy() {
```

Comment:
> Please also remember to set the width/height of the canvas to 0 before removing it, since that will cause (at least) Firefox to free graphics resources quicker

That is a good point, i didn't think about that. Thanks for pointing.